public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -536,7 +537,17 @@ export class DashboardModel {
     for (let i = 0; i < this.panels.length; i++) {
       const panel = this.panels[i];
       if (panel.repeat) {
-        this.repeatPanel(panel, i);
+        if (!affectedPanelIds) {
+          this.repeatPanel(panel, i);
+          continue;
+        }
+
+        if (
+          affectedPanelIds.includes(panel.id) ||
+          (panel.repeatPanelId && affectedPanelIds.includes(panel.repeatPanelId))
```

Comment:
think only repeating affected panels will be tricky, and could be unnecessary, not sure it will save any queries from being issued, if processRepeats runs with the same variable values no new queries should be issues anyway. 

in processRepeats we call cleanUpRepeats that removes all repeated panels so it expects everything to get re-added (repeated) every call

Reply:
Aha, that's true. I'll remove that. What about repeating rows? Are they called somewhere in the startRefresh path of the code?

Reply:
repeating rows is part of .this.repeatPanel (a row is a sort of full width panel)

Reply:
Aha, thank you!